internal/clients/security.go

```diff
@@ -138,3 +138,60 @@ func (a *ApiClient) DeleteElasticsearchRole(ctx context.Context, rolename string
 
 	return diags
 }
+
+func (a *ApiClient) PutElasticsearchRoleMapping(ctx context.Context, roleMapping *models.RoleMapping) diag.Diagnostics {
+	roleMappingBytes, err := json.Marshal(roleMapping)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	res, err := a.es.Security.PutRoleMapping(roleMapping.Name, bytes.NewReader(roleMappingBytes), a.es.Security.PutRoleMapping.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	defer res.Body.Close()
+	if diags := utils.CheckError(res, "Unable to put role mapping"); diags.HasError() {
+		return diags
+	}
+
+	return nil
+}
+
+func (a *ApiClient) GetElasticsearchRoleMapping(ctx context.Context, roleMappingName string) (*models.RoleMapping, diag.Diagnostics) {
+	req := a.es.Security.GetRoleMapping.WithName(roleMappingName)
```

Comment:
Can't we chain the WithContext call here?

Reply:
No, it's not possible...!